flat_mutation_reader.hh

```diff
@@ -141,6 +141,10 @@ public:
                     });
                 }
 
+                if constexpr (std::is_same_v<future<stop_iteration>, decltype(consumer(pop_mutation_fragment()))>) {
+                    return consumer(pop_mutation_fragment());
+                }
```

Comment:
I think the intent here was to make sure no exceptions are thrown from here, `futurize_invoke` catches exceptions and converts them to failed futures.

Reply:
Would it matter though? Repeat will also propagate the exception as a failed future, since it future_invokes this lambda

Reply:
I guess not.